lib/AST/Module.cpp

```diff
@@ -407,15 +426,41 @@ void ModuleDecl::removeFile(FileUnit &existingFile) {
   // Adjust for the std::reverse_iterator offset.
   ++I;
   Files.erase(I.base());
+  clearLookupCache();
 }
 
 #define FORWARD(name, args) \
   for (const FileUnit *file : getFiles()) \
     file->name args;
 
+SourceLookupCache &ModuleDecl::getSourceLookupCache() const {
+  if (!Cache) {
+    const_cast<ModuleDecl *>(this)->Cache =
+        llvm::make_unique<SourceLookupCache>(*this);
+  }
+  return *Cache;
+}
+
+static bool isParsedModule(const ModuleDecl *mod) {
+  // FIXME: If we ever get mixed modules that contain both SourceFiles and other
+  // kinds of file units, this will break; there all callers of this function should
+  // themselves assert that all file units in the module are SourceFiles when this
+  // function returns true.
```

Comment:
We do have this with SIB files. :-/ I guess if it's not tested it's not interesting, but…

Reply:
Yeah, that’s actually I’d tested, it’s why I bail out of the first file in SIL (if there’s a SIL file, there’s only one, and it’s the first). That deserves a comment.

Reply:
You can (in theory) have a plain Swift file mixed with SIB files.